ext/gd/gd.c

```diff
@@ -4058,114 +4155,44 @@ static gdIOCtx *create_stream_context(php_stream *stream, int close_stream) {
 	return ctx;
 }
 
-static gdIOCtx *create_output_context(void) {
-	gdIOCtx *ctx = ecalloc(1, sizeof(gdIOCtx));
-
-	ctx->putC = _php_image_output_putc;
-	ctx->putBuf = _php_image_output_putbuf;
-	ctx->gd_free = _php_image_output_ctxfree;
-
-	return ctx;
-}
-
-static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type, char *tn)
-{
-	zval *imgind;
-	zend_long quality = -1, basefilter = -1, speed = -1;
-	gdImagePtr im;
-	gdIOCtx *ctx = NULL;
-	zval *to_zval = NULL;
-
-	if (image_type == PHP_GDIMG_TYPE_GIF) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!", &imgind, gd_image_ce, &to_zval) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else if (image_type == PHP_GDIMG_TYPE_PNG) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!ll", &imgind, gd_image_ce, &to_zval, &quality, &basefilter) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else if (image_type == PHP_GDIMG_TYPE_AVIF) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!ll", &imgind, gd_image_ce, &to_zval, &quality, &speed) == FAILURE) {
-			RETURN_THROWS();
-		}
-	} else {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!l", &imgind, gd_image_ce, &to_zval, &quality) == FAILURE) {
-			RETURN_THROWS();
-		}
-	}
-
-	im = php_gd_libgdimageptr_from_zval_p(imgind);
+static gdIOCtx *create_output_context(zval *to_zval) {
```

Comment:
This should pass the arg_num to the function

Reply:
It seems a bit funny to pass the argument number here.. for all image output functions it's always the second one. But I'll add it anyway, maybe future functions might have it different.

Reply:
I would rather we do it properly now, then need to refactor it down the line.